### ISSUE TITLE
Crash with traitsui and pyface 6.1.0

### ISSUE THREAD

**Issue by FedeMiorelli** [enthought/mayavi#779]
After updating to latest traitsui 6.1.0 and pyface 6.1.0, clicking the pipeline toolbar button leads to a crash. No issues with versions 6.0.0 of both libraries.

mayavi==4.6.2
pyface==6.1.0
PyQt5==5.12.2
traits==5.1.1
traitsui==6.1.0

```

>>> Traceback (most recent call last):
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\pyface\ui\qt4\action\action_item.py", line 386, in _qt4_on_triggered
    action.perform(action_event)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\pyface\action\action.py", line 135, in perform
    self.on_perform()
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\mayavi-4.6.2-py3.7-win-amd64.egg\mayavi\core\ui\mayavi_scene.py", line 46, in show_engine
    return EngineRichView(engine=engine).scene_editing_view(scene=self)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\mayavi-4.6.2-py3.7-win-amd64.egg\mayavi\core\ui\engine_rich_view.py", line 164, in scene_editing_view
    return self.edit_traits()
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traits\has_traits.py", line 2024, in edit_traits
    args,
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\view.py", line 449, in ui
    ui.ui(parent, kind)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\ui.py", line 237, in ui
    self.rebuild(self, parent)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\toolkit.py", line 166, in ui_live
    ui_live.ui_live(ui, parent)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\ui_live.py", line 41, in ui_live
    _ui_dialog(ui, parent, BaseDialog.NONMODAL)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\ui_live.py", line 63, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\ui_base.py", line 278, in display_ui
    ui.owner.init(ui, parent, style)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\ui_live.py", line 203, in init
    self.add_contents(panel(ui), bbox)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\ui_panel.py", line 265, in panel
    panel = _GroupPanel(content[0], ui).control
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\ui_panel.py", line 558, in __init__
    self._add_splitter_items(content, splitter)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\ui_panel.py", line 638, in _add_splitter_items
    panel = self._add_items([item])
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\ui_panel.py", line 876, in _add_items
    editor.prepare(inner)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\editor.py", line 172, in prepare
    self.update_editor()
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\tree_editor.py", line 319, in update_editor
    self._expand_node(nid)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\tree_editor.py", line 515, in _expand_node
    self._append_node(nid, child_node, child)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\tree_editor.py", line 415, in _append_node
    return self._insert_node(nid, None, node, object)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\tree_editor.py", line 432, in _insert_node
    self._add_listeners(node, object)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\qt4\tree_editor.py", line 634, in _add_listeners
    node.when_label_changed(object, self._label_updated, False)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\tree_node.py", line 1516, in when_label_changed
    return object.tno_when_label_changed(self, listener, remove)
  File "C:\WinPython-64bit-3.7.2.0.new\python-3.7.2.amd64\lib\site-packages\traitsui\tree_node.py", line 1898, in tno_when_label_changed
    self._listener_cache[memo] = wrapped_listener
AttributeError: 'Scene' object has no attribute '_listener_cache'
```

**Comment by FedeMiorelli** [comment]
Seems related to this:
https://github.com/enthought/traitsui/pull/528

**Comment by corranwebster** [comment]
Thanks for reporting - this looks like a failure to initialize the new `_listener_cache` on a `TreeNodeObject`, and is a bug in the new TraitsUI release.

**Comment by corranwebster** [comment]
This should be fixed by https://github.com/enthought/traitsui/issues/556

**Comment by FedeMiorelli** [comment]
Thank you @corranwebster that was fast

**Comment by corranwebster** [comment]
Breaking Mayavi is a fairly significant problem for TraitsUI . :)

We will likely do a 6.1.1 release of TraitsUI in the next week or so (I want to wait a few more days to see if anything else comes up).

**Comment by prabhuramachandran** [comment]
Fixed in 6.1.1 I think so closing.
